File: java/src/main/java/io/fivexer/sdk/model/CreateNotificationSequence.java
package io.fivexer.sdk.model;

import io.fivexer.sdk.internal.Json;
import java.util.List;

/**
 * Input for creating a notification sequence.
 *
 * <p>Fluent builder: set what you need, then hand it to the client. {@link #toJson()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
public class CreateNotificationSequence {
    private String name;
    private List<NotificationSequenceStep> steps;
    private Boolean enabled;
    private List<String> filterTags;

    public CreateNotificationSequence(String name, List<NotificationSequenceStep> steps) {
        if (name == null) {
            throw new IllegalArgumentException("name is required");
        }
        if (steps == null) {
            throw new IllegalArgumentException("steps is required");
        }
        this.name = name;
        this.steps = steps;
    }

    public String getName() { return name; }
    public List<NotificationSequenceStep> getSteps() { return steps; }
    public Boolean getEnabled() { return enabled; }
    public List<String> getFilterTags() { return filterTags; }

    public CreateNotificationSequence enabled(Boolean enabled) { this.enabled = enabled; return this; }
    public CreateNotificationSequence filterTags(List<String> filterTags) { this.filterTags = filterTags; return this; }

    /** Serialise for the wire, omitting unset fields. */
    public String toJson() {
        return Json.write(this);
    }
}

File: php/src/Model/CreateNotificationSequence.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Input for creating a notification sequence.
 *
 * Fluent builder: set what you need, then hand it to the client. {@see self::toArray()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
final class CreateNotificationSequence
{
    public function __construct(
        private readonly string $name,
        /** @var list<NotificationSequenceStep> */
        private readonly array $steps,
        private ?bool $enabled = null,
        /** @var list<string> */
        private ?array $filterTags = null,
    ) {
    }

    public function enabled(bool $enabled): self
    {
        $this->enabled = $enabled;
        return $this;
    }

    /** @param list<string> $filterTags */
    public function filterTags(array $filterTags): self
    {
        $this->filterTags = $filterTags;
        return $this;
    }

    /**
     * Serialise for the wire, omitting unset fields.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'name' => $this->name,
            'steps' => Json::each($this->steps),
        ] + Json::compact([
            'enabled' => $this->enabled,
            'filterTags' => $this->filterTags,
        ]);
    }
}
